src/foam/u2/search/GroupBySearchView.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2.search',
+  name: 'GroupBySearchView',
+  extends: 'foam.u2.View',
+  requires: [
+    'foam.mlang.Constant',
+    'foam.mlang.predicate.True',
+    'foam.mlang.sink.Count',
+    'foam.mlang.sink.GroupBy',
+    'foam.u2.tag.Select'
+  ],
+
+  properties: [
+    {
+      name: 'view'
+    },
+    {
+      name: 'viewFactory',
+      value: { class: 'foam.u2.tag.Select' }
+    },
+    {
+      name: 'dao',
+      required: true,
+      postSet: function() {
+        this.updateDAO();
+      }
+    },
+    {
+      name: 'property',
+      required: true
+    },
+    {
+      name: 'name',
+      expression: function(property) {
+        return property.name;
+      }
+    },
+    {
+      class: 'Class2',
+      name: 'op',
+      value: 'foam.mlang.predicate.Eq'
+    },
+    {
+      name: 'predicate',
+      factory: function() {
+        return this.True.create();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'width',
+      value: 17
+    },
+    {
+      class: 'String',
+      name: 'label',
+      expression: function(property) {
+        return property.label;
+      }
+    }
+  ],
+
+  methods: [
+    function clear() {
+      this.view.data = '';
+    },
+    function initE() {
+      this.cssClass(this.myCls());
+      this.view = this.start(this.viewFactory);
+      this.view.end();
+
+      this.dao.on.sub(this.updateDAO);
+      this.view.data$.sub(this.updatePredicate);
+    },
+    function updatePredicate_(choice) {
+      var exists = typeof choice !== 'undefined' && choice !== '';
+      this.predicate = exists ? this.op$cls.create({
+        arg1: this.property,
+        arg2: this.Constant.create({ value: choice })
```

Comment:
I wonder if we should handle the adapting in the arg# adapt methods rather than in the external compile() method. Would mLangs easier to use directly.